summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala

```diff
@@ -144,8 +142,35 @@ private[storm] case class StormTopologyBuilder(options: Map[String, Options], jo
     case Some(props) =>
       registerShardedKeyValue[K, V](topology, node, provider, props.batcher, props.shards)
     case None =>
-      throw new IllegalStateException("Shouldn't happen, trying to register component which emits " +
-        "keyed values and don't have downstream summers.")
+      registerKeyValue[K, V](topology, node, provider)
```

Comment:
nit: I just noticed this method is becoming hard to read due to nested deconstructors, perhaps refactor "case Some(props) if props.allSummers =>" part into a separate method.